### ISSUE TITLE
"ENOENT: no such file or directory, open '/var/task/src/api/http/index_bg.wasm'"

### ISSUE THREAD

**Issue by bryantbiggs** [apollographql/federation#211]
Upgrading from @apollo/gateway v0.19.1 to v0.21.0 we are running into an issue with a missing wasm file:

```
{
    "errorType": "Error",
    "errorMessage": "ENOENT: no such file or directory, open '/var/task/src/api/http/index_bg.wasm'",
    "code": "ENOENT",
    "errno": -2,
    "syscall": "open",
    "path": "/var/task/src/api/http/index_bg.wasm",
    "stack": [
        "Error: ENOENT: no such file or directory, open '/var/task/src/api/http/index_bg.wasm'",
        "    at Object.openSync (fs.js:462:3)",
        "    at Object.readFileSync (fs.js:364:35)",
        "    at Object.<anonymous> (/var/task/src/api/http/webpack:/home/circleci/project/node_modules/@apollo/query-planner-wasm/dist/index.js:146:1)",
        "    at __webpack_require__ (/var/task/src/api/http/webpack:/webpack/bootstrap:19:1)",
        "    at Object.<anonymous> (/var/task/src/api/http/webpack:/home/circleci/project/node_modules/@apollo/gateway/dist/buildQueryPlan.js:5:1)",
        "    at __webpack_require__ (/var/task/src/api/http/webpack:/webpack/bootstrap:19:1)",
        "    at Object.<anonymous> (/var/task/src/api/http/webpack:/home/circleci/project/node_modules/@apollo/gateway/dist/index.js:22:1)",
        "    at __webpack_require__ (/var/task/src/api/http/webpack:/webpack/bootstrap:19:1)",
        "    at Object.<anonymous> (/var/task/src/api/http/webpack:/src/graphql/gateway.ts:2:1)",
        "    at __webpack_require__ (/var/task/src/api/http/webpack:/webpack/bootstrap:19:1)"
    ]
}
```
This is running in AWS lambda

**Comment by trevor-scheer** [comment]
Hey @bryantbiggs, thanks for opening this and letting us know! Would you mind sharing a reproduction that I can run to diagnose this?

My initial reaction is that this is webpack related. You may need to tweak your config to correctly load the wasm files. If you inspect the gateway's `node_modules` in the bundle you may see what's incorrect (something about how the wasm files are being `require`d most likely).

**Comment by bryantbiggs** [comment]
hey thanks @trevor-scheer yes I plan on putting up something - trying to get things rolled back and then I can get that put together

**Comment by trevor-scheer** [comment]
If you don't mind, I'd really appreciate if you would first do a bit of digging into your webpack config to see if this is something you can resolve yourself. I don't suspect I'm any more qualified to resolve a webpack config issue than you are since it's fairly outside of my wheelhouse.

If that's not fruitful, sharing your findings so far with a reproduction would be a great next step!

**Comment by manakao** [comment]
got same error

**Comment by dionescu-shipserv** [comment]
Same error happening here. I'm building a gateway application with Docker and then I'm coying the dist by doing `COPY dist/app usr/src/app`, excluding the node_modules.

**Comment by dai4** [comment]
Same error happening here. I'm building a gateway application with Docker and then I'm coying the dist by doing COPY dist/app usr/src/app, excluding the node_modules.

**Comment by thumphries** [comment]
Did my best to reproduce: https://gist.github.com/thumphries/6921be89fd70b43d184323333b42996d

Not quite the same as the OP but seems like the same issue. If you hack away the `require('util')` thing you get the same error.

I am also bundling for AWS Lambda, my repro uses `serverless-webpack`.
 
`query-planner-wasm` package seems to be built using `wasm-pack -t nodejs`, rather than `-t bundler`. Not sure if it's as simple as that, I tried to vendor and hit the limits of my webpack skills...

My workaround for now is to stick to `@apollo/gateway@0.19.1`, which bundles fine

**Comment by pmirandaarias** [comment]
Same here. I had to downgrande both:
```
    "@apollo/federation": "^0.19.1",
    "@apollo/gateway": "^0.19.1",
```
Lastest version of both give me the error.

**Comment by bryantbiggs** [comment]
@trevor-scheer will the repro that @thumphries provided suffice?

**Comment by Otard95** [comment]
Fairly certain #255 is related. I go through my, not so pretty, temporary solution in this [comment](https://github.com/apollographql/federation/issues/255#issuecomment-742433497)

**Comment by bryantbiggs** [comment]
I haven't tried it myself yet but based on #255 this issue should be resolved now with #270 and v0.24.4

**Comment by abernix** [comment]
Copying and pasting from https://github.com/apollographql/federation/issues/255#issuecomment-795328605:

> Please try `@apollo/gateway@0.24.4`, which (via `@apollo/query-planner@0.0.13`) includes `@apollo/query-planner-wasm@0.2.3` that shipped with the fix (and a couple follow-up commits) provided in #270. Would be happy to hear feedback about if this worked!

Please do report back how it goes and if this resolves things for this issue.  I can't promise there won't be any bundler configuration changes necessary, but this will hopefully facilitate that.
